src/pages/SupplierRequisitionPage.js

```diff
@@ -163,14 +179,22 @@ export class SupplierRequisitionPage extends React.Component {
    */
   refreshData = (newSearchTerm, newSortBy, newIsAscending) => {
     const { requisition } = this.props;
+    const { useThresholdMOS, orderType } = this.state;
 
-    this.updateDataFilters(newSearchTerm, newSortBy, newIsAscending);
-    const { searchTerm, sortBy, isAscending } = this.dataFilters;
+    if (newSearchTerm && newSortBy && newIsAscending) {
+      this.updateDataFilters(newSearchTerm, newSortBy, newIsAscending);
+    }
 
-    const data = requisition.items.filtered(
+    const { searchTerm, sortBy, isAscending } = this.dataFilters;
+    let data = requisition.items.filtered(
       'item.name BEGINSWITH[c] $0 OR item.code BEGINSWITH[c] $0',
       searchTerm
     );
+    if (useThresholdMOS) {
+      data = data.filter(requisitionItem =>
+        requisitionItem.item.isLessThanThresholdMOS(orderType.thresholdMOS)
```

Comment:
not a required change, but i wonder if realm can do this much quicker (and maybe we don't need that getter) don't know if this would work or not `data.filtered('stockOnHand < (monthlyUsage * $0)';thresholdMos)`

Reply:
I couldn't get a similar query to this to work. Using `data = data.filter(requisitionItem => requisitionItem.isLessThanThresholdMOS);` still 9095693bb8516259926361a7bf6d4ca464bdad4f